File: TodayFitComplete_Backend/src/main/java/com/SJY/TodayFitComplete_Backend/dto/member/MemberRegisterDto.java
package com.SJY.TodayFitComplete_Backend.dto.member;

import com.SJY.TodayFitComplete_Backend.entity.member.type.RoleType;
import com.SJY.TodayFitComplete_Backend.entity.member.Member;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 회원 가입 요청 DTO.
 */
@Getter
@Setter
@NoArgsConstructor
public class MemberRegisterDto {
    private String email;
    private String password;
    private String passwordCheck;
    private String username;

    public static Member ofEntity(MemberRegisterDto dto) {
        return Member.builder()
                .email(dto.getEmail())
                .password(dto.getPassword())
                .username(dto.getUsername())
                .roles(RoleType.USER)
                .build();
    }
}

File: TodayFitComplete_Frontend/app/src/main/java/kr/ac/kumoh/ce/s20190633/todayfitcomplete_frontend/Dto/Member/MemberRegisterDto.kt
package kr.ac.kumoh.ce.s20190633.todayfitcomplete_frontend.Dto.Member

// MemberRegisterDto 클래스: 사용자 회원 가입에 필요한 데이터를 나타내는 데이터 전송 객체 (DTO)
data class MemberRegisterDto(
    val email: String,        // 사용자 이메일 주소
    val password: String,     // 사용자 비밀번호
    val passwordCheck: String, // 비밀번호 확인을 위한 필드 (비밀번호와 일치해야 함)
    val username: String      // 사용자의 사용자 이름
)
